lmfit/parameter.py

```diff
@@ -458,52 +458,34 @@ def set(self, value=None, vary=None, min=None, max=None, expr=None):
             self._val = value
         if vary is not None:
             self.vary = vary
-        if min is not None:
-            self.min = min
-        if max is not None:
-            self.max = max
+        if min is None:
+            min = -inf
+        if max is None:
+            max = inf
+        self.min = min
+        self.max = max
 
     def _init_bounds(self):
         """make sure initial bounds are self-consistent"""
         # _val is None means - infinity.
+        if self.max is None:
+            self.max = inf
+        if self.min is None:
+            self.min = -inf
         if self._val is not None:
-            if self.max is not None and self._val > self.max:
+            if self.min > self.max:
+                self._min, self._max = self.max, self.min
+            if isclose(self.min, self.max, atol=1e-13, rtol=1e-13):
+                raise ValueError("Parameter '%s' has min == max" % self.name)
+
+            if self._val > self.max:
                 self._val = self.max
-            if self.min is not None and self._val < self.min:
+            if self._val < self.min:
                 self._val = self.min
-        elif self.min is not None and self._expr is None:
+        elif self._expr is None:
             self._val = self.min
-        elif self.max is not None and self._expr is None:
-            self._val = self.max
         self.setup_bounds()
 
-    def get_max(self):
-        return self._max
-
-    def set_max(self, val):
-        if val is None:
-            val = inf
-        self._max = val
-        if self.min > self.max:
-            self._min, self._max = self.max, self.min
-        if isclose(self.min, self.max, atol=1e-13, rtol=1e-13):
-            raise ValueError("Parameter '%s' has min == max" % self.name)
-
-    def get_min(self):
-        return self._min
-
-    def set_min(self, val):
```

Comment:
The checks in `set_min` should be carried out whenever `min` is changed. How will this be done if the properties are removed?